lib/money/currency.ex

```diff
@@ -423,12 +444,50 @@ defmodule Money.Currency do
     round(:math.pow(10, exponent))
   end
 
+  @spec number(Money.t() | String.t() | atom) :: integer | nil
+  @doc ~S"""
+  Returns the number of the currency or nil if it doesn’t exist.
+
+  ## Example:
+
+      iex> Money.Currency.number(:USD)
+      840
+      iex> Money.Currency.number(:WRONG)
+      nil
+  """
+  def number(%Money{currency: currency}),
+    do: number(currency)
+
+  def number(currency),
+    do: get(currency)[:number]
+
+  @spec number!(Money.t() | String.t() | atom) :: integer
+  @doc ~S"""
+  Returns the number of the currency.
+  An ArgumentError is raised if the currency doesn’t exist.
+
+  ## Example:
+
+      iex> Money.Currency.number!(:EUR)
+      978
+      iex> Money.Currency.exponent!(:WRONG)
+      ** (ArgumentError) currency WRONG doesn’t exist
+  """
+  def number!(currency),
+    do: number(currency) || currency_doesnt_exist_error(currency)
+
   defp convert_currency(currency) when is_binary(currency) do
     currency |> String.upcase() |> String.to_existing_atom() |> convert_currency
   rescue
     _ -> nil
   end
 
+  defp convert_currency(currency) when is_integer(currency) do
+    Map.fetch! @currency_numbers, currency
```

Comment:
We may not create a separate structure only for fetching the currency by number. Just use `Enum.find/2`
```elixir
case Enum.find(all_currencies(), fn {_, %{number: number}} -> number == currency end) do
  {_, cur} -> cur
  nil -> nil
end
```